packages/@aws-cdk/aws-dax/lib/cluster.ts

```diff
@@ -0,0 +1,348 @@
+   */
+  public readonly role: Role;
+
+  /**
+   * The underlying Cfn resource for the cluster
+   * @attribute
+   */
+  public readonly cluster: CfnCluster;
+
+  /**
+   * The endpoint at which the cluster can be accessed
+   * @attribute
+   */
+  public readonly clusterDiscoveryEndpoint: string;
+
+  /**
+   * The ARN of the cluster
+   * @attribute
+   */
+  public readonly clusterArn: string;
+
+  /**
+   * The name of the cluster
+   * @attribute
+   */
+  public readonly clusterName: string;
+
+  /**
+   * IAM policies associated with the cluster
+   */
+  protected policies: Policy[];
+
+  protected id: string;
+
+  constructor(scope: cdk.Construct, id: string, props: ClusterProps) {
+    super(scope, id);
+    this.id = id;
+    this.vpc = props.vpc;
+    this.clusterName = props.clusterName;
+
+    // Define a group for telling Elasticache which subnets to put cache nodes in.
+    const subnetGroup = new CfnSubnetGroup(this, `${id}-subnet-group`, {
+      description: `List of subnets used for dax cache ${id}`,
+      subnetIds: this.vpc.privateSubnets.map(subnet => subnet.subnetId),
+    });
+
+    // The security group that defines network level access to the cluster
+    this.securityGroup = new SecurityGroup(this, `${id}-security-group`, { vpc: this.vpc });
+
+    this.connections = new Connections({
+      securityGroups: [this.securityGroup],
+      defaultPort: new Port({
+        protocol: Protocol.TCP,
+        fromPort: 8111,
+        toPort: 8111,
+        stringRepresentation: 'TCP',
+      }),
+    });
+
+    this.role = new Role(this, `${id}-role`, {
+      assumedBy: new ServicePrincipal('dax.amazonaws.com'),
+      description: 'Role that allows DynamoDB Accelerator to talk to DynamoDB',
+    });
+
+    // The parameters for this cluster
+    let parameterGroup = props.parameterGroup;
+
+    if (!parameterGroup) {
+      parameterGroup = new ParameterGroup(this, '${id}-parameters', {
+        parameters: {
+          'query-ttl-millis': cdk.Duration.minutes(5).toMilliseconds().toString(),
+          'record-ttl-millis': cdk.Duration.minutes(5).toMilliseconds().toString(),
+        },
+      });
+    }
+
+    // The cluster resource itself.
+    this.cluster = new CfnCluster(this, `${id}-cluster`, {
+      iamRoleArn: this.role.roleArn,
+      nodeType: 'dax.' + props.instanceType.toString(),
+      replicationFactor: props.replicationFactor ? props.replicationFactor : 3,
+      subnetGroupName: subnetGroup.ref,
+      parameterGroupName: parameterGroup.parameterGroupName,
+      securityGroupIds: [
+        this.securityGroup.securityGroupId,
+      ],
+    });
+
+    this.clusterDiscoveryEndpoint = this.cluster.attrClusterDiscoveryEndpoint;
+    this.clusterArn = this.cluster.attrArn;
+
+    this.policies = [];
+  }
+
+  /**
+   * Add a table to the DAX cluster. This adds a policy on the same stack
+   * as the table while allows the IAM role of the cluster to have read/write
+   * access to the table.
+   */
```

Comment:
```suggestion
  /**
   * Add a table to the DAX cluster. This adds a policy on the same stack
   * as the table which allows the IAM role of the cluster to have read/write
   * access to the table.
   */
```

This also makes me wonder if there's value in adding a *read-only* table, too?